### PR TITLE
Handle leading zeroes in `rapids_cmake_write_version_file`

### PR DESCRIPTION
**Pull request by robertmaynard**
This removes the chance of generating octal values in C++ when the input looks like `02.09.08`

### CODE REVIEW

**Comment by robertmaynard** [comment]
I went with the `math(EXPR RAPIDS_WRITE_MINOR "${PROJECT_VERSION_MINOR} + 0" OUTPUT_FORMAT DECIMAL)` approach as it doesn't have issues with values like `00` which would get converted to an empty string.

File: testing/cmake/write_version-leading-zeroes/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#=============================================================================
+# Copyright (c) 2018-2021, NVIDIA CORPORATION.
```
Comment (dantegd) [inline testing/cmake/write_version-leading-zeroes/CMakeLists.txt:2]: The most nitpicky nitpick: All these copyright header years should be only 2021 I think

File: testing/cmake/write_version-relative/CMakeLists.txt [hunk 2]
```diff
@@ -17,7 +17,7 @@ include(${rapids-cmake-dir}/cmake/write_version_file.cmake)
 
 cmake_minimum_required(VERSION 3.20)
 
-project(DEMO VERSION 3.2.1 LANGUAGES NONE)
+project(DEMO VERSION 3.2.0 LANGUAGES NONE)
```
Comment (dantegd) [inline testing/cmake/write_version-relative/CMakeLists.txt:20]: I think the only case not being tested is if this is 3.2, that the patch version would be 0

**Review by dantegd — APPROVED** [review]
Code looks great and testing it locally it seems to work very well! Just had two super minor questions

File: testing/cmake/write_version-leading-zeroes/CMakeLists.txt (hunk 1 above)
Comment (robertmaynard) [reply testing/cmake/write_version-leading-zeroes/CMakeLists.txt:2]: I followed the pattern that existed in cudf and raft where a copyright range existed. But at the minimum the copyright range is wrong ( should be 2021-2021? ), or do we change it next year to 2021-2022?

Comment (robertmaynard) [reply testing/cmake/write_version-leading-zeroes/CMakeLists.txt:2]: https://github.com/rapidsai/rapids-cmake/issues/28

**Pull request merged**